blues/sparrow-reference-web-app
# bug: better error handling if db server is not running

# Problem Context

Currently, if the Postgres DB is not running (or unreachable in some form or fashion), the UI gives the user a fairly generic message something's wrong, but it's not exactly sure what. Let's give them some better info to debug with and fix the issue.

## Changes

Added a new error handling `try/catch` block to the Prisma DB call that fires off as soon as the app loads, and a new error message for when the DB server can't be reached to inform the user of what the issue might be.

## Screenshot (if applicable)

![Screen Shot 2022-04-19 at 4 39 52 PM](https://user-images.githubusercontent.com/20400845/164091875-f6376d25-e831-4700-99b4-df4dd0f80af7.png)

## Testing

Unit / integration / e2e tests?

No new automated tests.

Steps to test manually?

1. Fire up the Sparrow app locally without doing any of the setup necessary for a connection to a locally running Postgres DB instance.
2. Go to http://localhost:4000/dev
3. See the error message in the browser warning that there's currently no database connection.
4. Go through the process of setting up a Ngrok tunnel connection to a Postgres server
5. Stop and restart the Sparrow app again
6. Go back to the same URL and see data populating into the page

Any other sorts of testing notes to include?

## Any other related PRs

n/a

## Ticket(s)

https://www.pivotaltracker.com/story/show/181911076
